strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/test/java/org/carlspring/strongbox/artifact/coordinates/PypiArtifactCoordinatesTest.java

```diff
@@ -0,0 +1,274 @@
+        parsedPackages.add(new ArrayList<>(Arrays.asList("cryptography", "2.6.1", "0", "cp27", "cp27mu", "manylinux1_x86_64", "whl")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("cryptography", "2.6.1", null, "cp34", "cp34m", "win32", "whl")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("protobuf", "3.7.1", null, "cp27", "cp27m", "macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64", "whl")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("protobuf", "3.7.1", "1", "cp35", "cp35m", "win32", "whl")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("protobuf", "3.7.1", null, "cp37", "cp37m", "macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64", "whl")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("virtualenv", "16.4.3", null, "py2.py3", "none", "any", "whl")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("coverage", "4.5.3", null, "cp26", "cp26m", "macosx_10_12_x86_64", "whl")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("coverage", "4.5.3", "5", "cp34", "cp34m", "win32", "whl")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("coverage", "4.5.3", "2", "cp36", "cp36m", "manylinux1_x86_64", "whl")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("docker", "3.7.2", null, "py2.py3", "none", "any", "whl")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("bcrypt", "3.1.6", null, "cp27", "cp27m", "macosx_10_6_intel", "whl")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("bcrypt", "3.1.6", null, "cp34", "cp34m", "manylinux1_i686", "whl")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("bcrypt", "3.1.6", null, "cp37", "cp37m", "win_amd64", "whl")));
+
+        parsedPackages.add(new ArrayList<>(Arrays.asList("distribution", "1.0", null, null, null, null, "tar.gz")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("example_pkg_your_username", "1.5.0", null, null, null, null, "tar.gz")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("someproject", "1.5.0", null, null, null, null, "tar.gz")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("six", "1.12.0", null, null, null, null, "tar.gz")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("futures", "3.2.0", null, null, null, null, "tar.gz")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("pytz", "2018.9", null, null, null, null, "tar.gz")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("numpy", "1.16.2", null, null, null, null, "tar.gz")));
+        parsedPackages.add(new ArrayList<>(Arrays.asList("bcrypt", "3.1.6", null, null, null, null, "tar.gz")));
+
+
+
+    }
+
+    /**
+     * Tests the constructor of PypiArtifactCoordinates with real Python package names.
+     * Simply add more pre-parsed filenames to parsedPackages array to test additional filenames.
+     */
+    @Test
```

Comment:
we better try to use `@RepeatedTest` annotation here

Reply:
Thanks! I added this annotation.

Reply:
+1: Yeah, I think this would be a very good idea!

Reply:
we need to do it like [this](https://github.com/strongbox/strongbox/pull/1346/commits/e52b98eabd6d420c212cf676e508accb1cfec566#r296156202)